Language/Expressions/Constants/Constant_Contexts/constant_context_A05_t02.dart

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Let e be an expression; e occurs in a constant context iff one of
+/// the following applies:
+/// ...
+/// • e is an immediate subexpression of an expression e0 which occurs in a
+///   constant context, where e0 is not a function literal
+///
+/// @description Checks that it is a compile-time error if `e` is an immediate
+/// subexpression of an expression `e0` which occurs in a constant context but
+/// `e0` is a function literal
+/// @author sgrekhov22@gmail.com
+
+class C {
+  final val;
+  const C(this.val);
+}
+
+main() {
+  const c1 = (){C(1);};
```

Comment:
This is actually the initializing expression, not a subexpression of any other expression.

Perhaps?:
```suggestion
  const c1 = C((){});
```
`c2` needs a similar adjustment, but `c3` is fine (except that perhaps the `^^^` should be adjusted).

Reply:
Changed to `C((){C(1);})`. According to the description `C(1)` must be a subexpression of a function literal, therefore I added it to the body